From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: byquanton <32410361+byquanton@users.noreply.github.com>
Date: Tue, 25 Apr 2023 00:49:31 +0200
Subject: [PATCH] Dont close Slime Plugin Classloader


diff --git a/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java b/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java
index 7572a0bf6614b02be3cbccc7b86e52ee1b8df621..0164e713678e627c955c36b3b9f56ca431fe96b7 100644
--- a/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java
+++ b/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java
@@ -255,8 +255,10 @@ public final class PluginClassLoader extends URLClassLoader implements io.paperm
             // Paper end
             super.close();
         } finally {
-            this.closed = true; // Pufferfish
-            jar.close();
+            if (this.plugin == null || !this.plugin.getPluginMeta().getProvidedPlugins().contains("SlimeWorldManager")) {
+                this.closed = true; // Pufferfish
+                jar.close();
+            }
         }
     }
 
